src/Smalot/PdfParser/Encoding.php

```diff
@@ -145,6 +145,12 @@ protected function getEncodingClass(): string
     {
         // Load reference table charset.
         $baseEncoding = preg_replace('/[^A-Z0-9]/is', '', $this->get('BaseEncoding')->getContent());
+
+        // Check for empty BaseEncoding field value
+        if (!\is_string($baseEncoding) || 0 == \strlen($baseEncoding)) {
```

Comment:
Just to make sure: if `$baseEncoding` is `null` or an `[ ]`, only the first part of the if-clauses is being used and the second part is not used, correct? The reason why I am asking, `strlen` triggers a warning or type error, if it receives a non-string as parameter.

Reply:
Yes. If `!is_string()` is `true`, the second part of the expression isn't tested since it isn't needed. The only way the `strlen()` function is reached is if the first expression evaluated to `false` which means it *must* be a string.